### ISSUE TITLE
Boundary Event Triggers Event Sub Process

### ISSUE THREAD

**Issue by sombrek** [bpmn-io/bpmn-js-token-simulation#181]
### Describe the Bug

When clicking an error or escalation boundary event, an event sub process is triggered instead.

### Steps to Reproduce

Have a look at the [full capacity demo](https://bpmn-io.github.io/bpmn-js-token-simulation/modeler.html?pp=1&diagram=https%3A%2F%2Fraw.githubusercontent.com%2Fbpmn-io%2Fbpmn-js-token-simulation%2Fmaster%2Fexample%2Fresources%2Fall.bpmn&e=1)

1. Go to the TEST ESCALATION pool (upper left)
2. Activate the Pause for "SUB"
3. Start the START event
4. When the token is waiting on SUB, click Play on BOUNDARY_E

### Expected Behavior

The boundary event should be triggered and token should flow to the "TEST FAILED" end event.
However, EVT_E is triggered instead.

### Environment

- Library version: 0.34.2

**Comment by nikku** [comment]
@sombrek This is expected behavior, per BPMN spec, I think.

Clicking the boundary will _trigger the event_, and not "follow the boundary outgoing path". Triggering the event will _always, per BPMN spec_ first trigger respective event-subprocesses. These can recover from the error.

I may be wrong and suggest you to provide a minimal test case of the wrong behavior you're seeing.

**Comment by sombrek** [comment]
@nikku I think this time I'm wrong.

Your explanation is correct and I'd expect the same behavior from a real process.

Here's the reduced model: [diagram.bpmn.txt](https://github.com/user-attachments/files/16645105/diagram.bpmn.txt)
The steps to reproduce 2-4 apply. When triggering the boundary event, the event subprocess starts instead.

I expected that either I cannot trigger the boundary event at all, since it cannot happen per spec; or that I can trigger the event and follow the exeption flow, since that happens for most other boundary events. I wasn't expecting to trigger an event somewhere else, but I now understand why it happens.

**Comment by nikku** [comment]
> @nikku I think this time I'm wrong.

That is ok, sometimes :slightly_smiling_face:.

> I wasn't expecting to trigger an event somewhere else, but I now understand why it happens.

The simulation is [a real engine](https://page.camunda.com/ccs2022-bpmn-js-token-simulation), and triggering a "boundary event" simply triggers the event that "it waits for". If you don't know this, then you may be surprised at times. The alternative would be to "mock too much" and not be able to explain some of BPMN's concepts around cancelation scopes, and recovery.

I'm closing this issue as won't fix.
